hnn_core/gui/gui.py

```diff
@@ -1254,33 +1253,38 @@ def _get_evoked_widget(name, layout, style, location, data=None,
     return drive, drive_box
 
 
-def _get_tonic_widget(name, layout, style, data=None):
-
-    default_data = {
+def _get_tonic_widget(name, tstop_widget, layout, style, data=None):
+    cell_types = ['L2_basket', 'L2_pyramidal', 'L5_basket', 'L5_pyramidal']
+    default_values = {
         'amplitude': 0,
         't0': 0,
-        'tstop': 0
+        'tstop': tstop_widget.value
     }
+    t0 = default_values['t0']
+    tstop = default_values['tstop']
+    default_data = {cell_type: default_values for cell_type in cell_types}
 
-    cell_types = ['L2_basket', 'L2_pyramidal', 'L5_basket', 'L5_pyramidal']
-
-    if isinstance(data, dict):
-        default_data.update(data)
     kwargs = dict(layout=layout, style=style)
+    if isinstance(data, dict):
+        default_data = _update_nested_dict(default_data, data)
 
     amplitudes = dict()
     for cell_type in cell_types:
-        if cell_type in data:
```

Comment:
`update_nested_dict()` gets rid of the need for this logic?

Reply:
Yep. The default_data dict is built on line 1265.  It is a nested dict by cell_type. Then update_nested_dict updates that structure with the data, which is also in that structure.